### PR TITLE
Try-except for better exceptions

### PR DESCRIPTION
**Pull request by samedii**
Try-except is fast as long as "except" does not happen so this should be fine for us. Really helpful messages but could maybe get too verbose to show "item".

Example:
```
  File "/home/richard/Documents/pytorch-datastream/datastream/dataset.py", line 135, in composed_fn
    raise Exception('\n'.join([
Exception: TypeError("<lambda>() missing 1 required positional argument: 'match_id'")
Above exception occured in the pipeline for
item:
('red', '4676460676')
file: /home/richard/Documents/league/venv/.guild/runs/729ed590e2054f13bfced64d4a4fb02f/.guild/sourcecode/league/problem/evaluate_datasets.py
source:
        .map(lambda side, match_id: (
            side,
            Match.from_disk(match_id),
        ))

```

### CODE REVIEW

**Comment by samedii** [comment]
This is very nice when we have lambdas that are executed in by `.map`/`.starmap` but otherwise I feel that we risk confusing the user. Maybe we can inspect the exception and only show the extra information when it is helpful

File: datastream/dataset.py [hunk 1]
```diff
@@ -125,12 +126,31 @@ def map(
         ... )[-1]
         4
         '''
+
+        def composed_fn(dataframe, index):
+            item = self.get_item(dataframe, index)
+            try:
+                return function(item)
+            except Exception as e:
+                item_text = str(item)
```
Comment (FelixAbrahamsson) [inline datastream/dataset.py:135]: Can we always guarantee that `item` implements `__str__`?

Comment (FelixAbrahamsson) [reply datastream/dataset.py:135]: Or I guess it doesn't matter since `str` can never crash, right?

Comment (samedii) [reply datastream/dataset.py:135]: Not sure. I think we should test this a bit more

Comment (Jim-Holmstroem) [reply datastream/dataset.py:135]: a general guideline
```
    The __repr__ is for developers. The essential purpose is to explain what the object is in an unambiguous way.

    The __str__ is for end users. The essential purpose is to explain what data the object contains in a readable manner.
```
`repr(.)` always exist (unless you explicitly remove it)

File: datastream/dataset.py [hunk 2]
```diff
@@ -125,12 +126,31 @@ def map(
...
+                if len(item_text) > 79:
```
Comment (Jim-Holmstroem) [inline datastream/dataset.py:136]: https://docs.python.org/3.4/library/textwrap.html#textwrap.shorten

**Pull request merged**